android/KMEA/app/src/main/java/com/tavultesoft/kmea/KMManager.java

```diff
@@ -647,6 +650,41 @@ public static Typeface getFontTypeface(Context context, String fontFilename) {
     return KeyboardPickerActivity.getKeyboardsList(context);
   }
 
+  public static boolean registerLexicalModel(HashMap<String, String> lexicalModelInfo) {
+    String pkgID = lexicalModelInfo.get(KMKey_PackageID);
+    String modelID = lexicalModelInfo.get(KMKey_LexicalModelID);
+    String languageID = lexicalModelInfo.get(KMKey_LanguageID);
+    String path = "file://" + getLexicalModelsDir() + pkgID + File.separator + modelID + ".model.js";
+
+    JSONObject modelObj = new JSONObject();
+    JSONArray languageJSONArray = new JSONArray();
+    try {
+      modelObj.put("id", modelID);
+      languageJSONArray.put(languageID);
+      modelObj.put("languages", languageJSONArray);
+      modelObj.put("path", path);
+    } catch (JSONException e) {
+      Log.e(TAG, "Invalid lexical model to register");
+      return false;
+    }
+
+    // Use single quotes in JSONobject for javascript call
+    String model = String.valueOf(modelObj);
+    model = model.replaceAll("\"", "'");
```

Comment:
This bothers me: it may come back to bite us (what if we had an embedded single quote in a string, for example). Why do we need to do this?

Reply:
w/o the substitution, KMW was throwing `Uncaught Error: Script error.`

I can follow precedent of escaping single-quoted names for the JS calls

Reply:
So why was it doing that? It seems wrong that single quotes would need escaping. Would it be possible to use [`evaluateJavascript`](https://developer.android.com/reference/android/webkit/WebView.html#evaluateJavascript(java.lang.String,%20android.webkit.ValueCallback%3Cjava.lang.String%3E)) instead?

Reply:
The reason:  keymanapp/lexical-models#15.